### DISCUSSION TITLE
Queue or playlist: am I doing it wrong?

### DISCUSSION THREAD

**Discussion by daveriedstra** [Taiko2k/Tauon#924] [Q&A] [answered]
I recently discovered Tauon and it's really great, thanks for it!

So, I'm using Tauon by having a main playlist for my music library, which I browse and use to queue up tracks / albums to play. This is generally fine, except that it seems like the queue is missing some features that are meant for playing back directly from a playlist: shuffle and repeat don't work, and when the queue is done the next item from my main library playlist starts playing. (This can be really disruptive, eg, moving from something really quiet into something much louder.)

This makes me think I'm using Tauon "wrong" and that the "right" or "intended" way to do this is to build up a new playlist instead of adding to the queue, except that there's also friction with this approach (there's no quick "add to playlist" action or in the right-click context menu for albums or tracks).

So I guess I'm wondering if I'm missing the ideal way to use Tauon. What I think I'm looking for is either a) extension of the playlist features to the queue (stop at end, shuffle, repeat), or b) better support for moving tracks between playlists and creating transient playlists to act as a queue, but I think I might not be doing it right...? What should I be doing instead?

**Comment by Taiko2k** [comment, marked as answer]
You're mostly right, its just the intended way of quickly adding tracks to a playlist is to use drag and drop.

**Reply by daveriedstra** [reply]
Oh! I didn't even realize that was a thing. I also just noticed I could convert a queue to a playlist. I'll try living with these for a while and see if that helps. Thanks!

**Reply by daveriedstra** [reply]
Yep, this is great. I was doing it wrong 🙃 Thanks!

**Comment by Solarunit** [comment]
>there's no quick "add to playlist" action or in the right-click context menu for albums or tracks

There is, try Middle click on the song :wink:
Also Middle click on the song in the Queue removes it from the list.

**Reply by Taiko2k** [reply]
Middle click adds it to the queue but OP wanted to quick add to a playlist.

**Reply by daveriedstra** [reply]
Thanks for mentioning it though, I wasn't aware of this feature
